app/controllers/actions/AuthActionHelper.scala

```diff
@@ -36,4 +39,13 @@ trait AuthActionHelper {
   def authorisedDataRequestAsync(ern: String, draftId: String)(block: DataRequest[_] => Future[Result]): Action[AnyContent] =
     authorisedWithData(ern, draftId).async(block)
 
+  def ifCanUseDraftTemplates(ern: String, draftId: String)(block: Future[Result])(implicit appConfig: AppConfig): Future[Result] = {
```

Comment:
The `block` function needs to be passed by `reference` and not by `value`. Otherwise, it's value will be executed on calling the method rather only when referenced within the function.

To do this, add ` =>` before the `type` in the method signature:
 
i.e.:
`block: => Future[Result]`